### PR TITLE
Feature: Adds support for filling the bar with icons from the official weather card

### PR DESCRIPTION
**Pull request by scinos**
**Note**: The goal of this PR is to get initial feedback for the idea. If we are happy with the approach, I'll add tests, clean up the code and deal with corner cases.

---

## What problem does this PR solve?

I have this card in a wallboard. However, the icons are too small to be viewed across the room, and I can never remember which color represents each weather condition. 

## Proposed solution

I'd like to have a view more similar to the Weather app in Android: instead of using bars, display an icon for each hour/segment. Optionally, use more colorful icons.

I've introduced two new options:
- `icon_fill`: instead of displaying a bar of length N for each weather condition and an icon in the center of the bar, it will add N `<span>` tags, each with the same icon.
- `official_icons`: instead of using `mdi:*` icons, it will use SVGs from the official [Weather Forecast card](https://www.home-assistant.io/dashboards/weather-forecast/). Unfortunately I don't know how to grab those icons from the host HomeAssistant instance, so I had to copy the code over from their repo.

This is how it looks like:

image

## Problems/questions

- Currently, setting `icon_fill:false` and `official_icons:true` will result in a broken UI, with huge icons. Need to update my CSS skills to solve it.
- I'm removing the condition class (eg: `class="cloudy"`) to hide the bar colors. Arguably, this class should stay and we should hide the color using something else. Open to ideas.
- Not a big fan of the names `icon_fill` or `official_icons`, open to suggestions.
- I don't fully understand the `max-width` property of `condition-icon` class. I know it has something to do with hiding the icon if the space is too small, but not sure how it is computed and the best way to override it when `icon_fill:true`.

PS: Cheers from a fellow ex-Atlassian :D

### CODE REVIEW

File: src/lib/homeassistant-weather-icons/index.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
+import { SVGTemplateResult, svg, css } from "lit";
+
+const cloudyStates = new Set<string>([
```
Comment (scinos) [inline src/lib/homeassistant-weather-icons/index.ts:3]: I copied this as-is. Do we really need this many states? Where can I read more about this topic?

File: src/weather-bar.ts [hunk 2]
```diff
@@ -237,6 +273,11 @@ export class WeatherBar extends LitElement {
       max-width: max(0px, calc((100% - 40px) * 999));
       overflow: hidden;
     }
+    .condition-icon-big {
```
Comment (scinos) [inline src/weather-bar.ts:276]: I don't like this class, it introduces some unnecessary duplication. But not sure if there is a better way to override `max-width` from `.condition-icon`.

**Comment by scinos** [comment]
@decompil3d would you have some time to have a look to this PR? If you are ok with the feature, I can continue working on it (mainly add tests).

File: README.md [hunk 3]
```diff
@@ -77,6 +77,9 @@ decimal by 1). Otherwise, the integration may complain of a duplicate unique ID.
...
 | `round_temperatures`             | bool             | **Optional** | Whether to round temperatures to the nearest whole number                                                                                                       | `false`             |
 | `hide_bar`                       | bool             | **Optional** | Whether to hide the bar itself                                                                                                                                  | `false`             |
+| `icon_fill`                      | bool             | **Optional** | Whether to repeat the icon inside the bar (one icon per segment)                                                                                                | `false`             |
+| `official_icons`                 | bool             | **Optional** | Whether to use the icons from the official Weather Forecast card                                                                                                | `false`             |
```
Comment (decompil3d) [inline README.md:81]: I am ok with the `icon_fill` option. But I don't think that these alternate icons should go in. It adds unnecessary bloat to the package as most users won't use these other icons. I'd suggest using card-mod and CSS to hide the default icons and show your own in place using the CSS `content`rule.

File: src/hourly-weather.ts [hunk 4]
```diff
@@ -313,12 +320,14 @@ export class HourlyWeatherCard extends LitElement {
 
     const entityId: string = config.entity;
     const state = this.hass.states[entityId];
+    const dayTime = this.getDayNightTimes();
```
Comment (decompil3d) [inline src/hourly-weather.ts:323]: This won't work for days beyond the current day. Please remove.

File: src/weather-bar.ts [hunk 5]
```diff
@@ -212,16 +259,17 @@ export class WeatherBar extends LitElement {
       --color-windy: var(--color-sunny);
       --color-windy-variant: var(--color-sunny);
       --color-exceptional: #ff9d00;
+      --color-night: #333;
     }
     .bar {
-      height: 30px;
+      height: 48px;
```
Comment (decompil3d) [inline src/weather-bar.ts:265]: Please do not change this - since we won't support alternate icons, this should stay as-is.

**Review by decompil3d — CHANGES_REQUESTED** [review]
Thanks for the contribution. While I'm ok with the `icon_fill` option and associated `segment_spacing` option, I don't like the alternate icon set. It adds unnecessary bloat to the package. Also, the daytime/night calculation won't work for days beyond the current day, so that needs to be rethought or removed.

Ping me when you're ready for another review on the core code change. Once we're aligned on the main code changes, you can move onto writing tests.

Thanks!
